megamek/src/megamek/common/weapons/WeaponHandler.java

```diff
@@ -171,9 +171,9 @@ protected int getLargeCraftHeat(Entity entity) {
                     int loc = prevWeapon.getLocation();
 
                     // create an array of booleans of locations
-                    boolean[] usedFrontArc = new boolean[ae.locations()];
-                    boolean[] usedRearArc = new boolean[ae.locations()];
-                    for (int i = 0; i < ae.locations(); i++) {
+                    boolean[] usedFrontArc = new boolean[entity.locations()];
+                    boolean[] usedRearArc = new boolean[entity.locations()];
+                    for (int i = 0; i < entity.locations(); i++) {
```

Comment:
```suggestion
                    boolean[] usedFrontArc = new boolean[ae.locations()];
                    boolean[] usedRearArc = new boolean[ae.locations()];
                    for (int i = 0; i < ae.locations(); i++) {
```

this should not be in this PR, I am reverting it

Reply:
I was looking at that recently so it seemed perfectly natural -_-;

Reply:
I believe it is correct and should be entity, but anyway... I was not working with this file, so it should not be pushed